test/vitest/healthcareProfessionalsStore.spec.ts

```diff
@@ -0,0 +1,116 @@
+    })
+
+    it('should return the correct names based on the locale', () => {
+        const healthcareProfessionalsStore = useHealthcareProfessionalsStore()
+        const mockHealthcareProfessional: HealthcareProfessional = {
+            id: '123',
+            acceptedInsurance: [Insurance.InsuranceNotAccepted],
+            createdDate: '2025-04-14',
+            updatedDate: '2025-04-15',
+            degrees: [Degree.Md],
+            facilityIds: ['facility1'],
+            names: [{ firstName: 'John', lastName: 'Doe', middleName: 'Deer', locale: Locale.EnUs },
+                    { firstName: 'ジョン', lastName: 'ドウ', middleName: 'ディア', locale: Locale.JaJp }],
+            specialties: [Specialty.Neurology],
+            spokenLanguages: [Locale.EnUs, Locale.JaJp]
+        }
+
+        const localeStore = useLocaleStore()
+
+        localeStore.locale.code = Locale.EnUs
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .firstName).to.equal('John')
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .middleName).to.equal('Deer')
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .lastName).to.equal('Doe')
+
+        localeStore.locale.code = Locale.JaJp
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .firstName).to.equal('ジョン')
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .middleName).to.equal('ディア')
+        expect(healthcareProfessionalsStore.displayChosenLocaleForHealthcareProfessional(mockHealthcareProfessional)
+            .lastName).to.equal('ドウ')
+    })
+
+    it('should reset the fields in the healthcare professional section', () => {
+        const healthcareProfessionalsStore = useHealthcareProfessionalsStore()
+
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .acceptedInsurance = [Insurance.InsuranceNotAccepted]
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .degrees = [Degree.Md]
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .facilityIds = ['facility1']
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .names = [{ firstName: 'John', lastName: 'Doe', middleName: 'Deer', locale: Locale.EnUs }]
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .specialties = [Specialty.Neurology]
+        healthcareProfessionalsStore.createHealthcareProfessionalSectionFields
+            .spokenLanguages = [Locale.EnUs]
+
+        healthcareProfessionalsStore.resetCreateHealthcareProfessionalFields()
```

Comment:
For a test like this it's common practice to `expect` what all of the fields are before calling the reset function. This is to confirm that the fields have values before they are reset. For example, you'd have:

```ts
expect(healthcareProfessionalsStore.createHealthcareProfessionalSectionFields.acceptedInsurance).to.deep.equal([Insurance.InsuranceNotAccepted])

healthcareProfessionalsStore.resetCreateHealthcareProfessionalFields()

expect(healthcareProfessionalsStore.createHealthcareProfessionalSectionFields.acceptedInsurance).to.deep.equal([])
```